staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go

```diff
@@ -46,15 +46,36 @@ import (
 	"k8s.io/kubectl/pkg/validation"
 )
 
+// ApplyFlags directly reflect the information that CLI is gathering via flags.  They will be converted to Options, which
+// reflect the runtime requirements for the command.  This structure reduces the transformation to wiring and makes
+// the logic itself easy to unit test
+type ApplyFlags struct {
+	Factory cmdutil.Factory
```

Comment:
See if you can use `genericclioptions.RESTClientGetter` rather than this factory. I'd like to get rid of this factory in the long run.

Reply:
sounds good, thx